File: OCTOBER-2025/15-10-2025/p3350.java
class Solution {
    public int maxIncreasingSubarrays(List<Integer> nums) {
        int n = nums.size();
        int inc = 1;
        int prevInc = 0;
        int ans = Integer.MIN_VALUE;

        for(int i = 1; i < n; i++){
            if(nums.get(i) > nums.get(i-1)){
                inc++;
            }
            else{
                prevInc = inc;
                inc = 1;
            }
            if(prevInc >= inc) ans = Math.max(ans, inc);
            else ans = Math.max(ans, inc/2);  
        }
        return ans;
    }
}

File: OCTOBER-2025/15-10-2025/p3350.cpp
class Solution {
public:
    int maxIncreasingSubarrays(vector<int>& nums) {
        int n = nums.size();
        vector<int> v(n, 1);

        for(int i=0; i<n; ++i) {
            if(i > 0 && nums[i] > nums[i-1]) {
                v[i] = v[i-1] + 1;
            }
        }

        int ans = 1;
        for(int i=n-1; i>=0; --i) {
            int len = v[i];
            if(i >= len) {
                int a1 = min(len, v[i-len]);
                ans = max(ans, a1);
            }
            ans = max(ans, len / 2);
        }

        return ans;
    }
};